mm2src/gui_storage/src/account/storage/sqlite_storage.rs

```diff
@@ -0,0 +1,580 @@
+            sql_delete_coins.delete()?;
+        }
+
+        transaction.commit()?;
+        Ok(())
+    }
+
+    /// Updates the given `account_id` account by applying the `update_cb` callback to an `SqlUpdate` SQL builder.
+    fn update_account<F>(conn: &Connection, account_id: AccountId, update_cb: F) -> AccountStorageResult<()>
+    where
+        F: FnOnce(&mut SqlUpdate) -> SqlResult<()>,
+    {
+        // First, check if the account exists.
+        if !Self::account_exists(conn, &account_id)? {
+            return MmError::err(AccountStorageError::NoSuchAccount(account_id));
+        }
+
+        let mut sql_update = SqlUpdate::new(conn, account_table::TABLE_NAME)?;
+        update_cb(&mut sql_update)?;
+
+        let (account_type, account_idx, device_pubkey) = account_id.to_sql_tuple();
+        sql_update
+            .and_where_eq(account_table::ACCOUNT_TYPE, account_type)?
+            .and_where_eq(account_table::ACCOUNT_IDX, account_idx)?
+            .and_where_eq_param(account_table::DEVICE_PUBKEY, device_pubkey)?;
+
+        sql_update.update()?;
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl AccountStorage for SqliteAccountStorage {
+    async fn init(&self) -> AccountStorageResult<()> {
+        let mut conn = self.lock_conn_mutex()?;
+        let transaction = conn.transaction()?;
+
+        SqliteAccountStorage::init_account_table(&transaction)?;
+        SqliteAccountStorage::init_account_coins_table(&transaction)?;
+        SqliteAccountStorage::init_enabled_account_table(&transaction)?;
+
+        transaction.commit()?;
+        Ok(())
+    }
+
+    async fn load_accounts(&self) -> AccountStorageResult<BTreeMap<AccountId, AccountInfo>> {
+        let conn = self.lock_conn_mutex()?;
+        Self::load_accounts(&conn)
+    }
+
+    async fn load_accounts_with_enabled_flag(
+        &self,
+    ) -> AccountStorageResult<BTreeMap<AccountId, AccountWithEnabledFlag>> {
+        let conn = self.lock_conn_mutex()?;
+        let enabled_account_id = AccountId::from(Self::load_enabled_account_id_or_err(&conn)?);
+
+        let mut found_enabled = false;
+        let accounts = Self::load_accounts(&conn)?
+            .into_iter()
+            .map(|(account_id, account_info)| {
+                let enabled = account_id == enabled_account_id;
+                found_enabled |= enabled;
+                Ok((account_id, AccountWithEnabledFlag { account_info, enabled }))
+            })
+            .collect::<AccountStorageResult<BTreeMap<_, _>>>()?;
+
+        // If `AccountStorage::load_enabled_account_id_or_err` returns an `AccountId`,
+        // then corresponding account must be in `AccountTable`.
+        if !found_enabled {
+            return MmError::err(AccountStorageError::unknown_account_in_enabled_table(
+                enabled_account_id,
+            ));
+        }
+        Ok(accounts)
+    }
+
+    async fn load_enabled_account_id(&self) -> AccountStorageResult<EnabledAccountId> {
```

Comment:
This seems to be used only in testing. Should we keep this as part of the trait? Why not use
```
        let conn = self.lock_conn_mutex()?;
        Self::load_enabled_account_id_or_err(&conn)
```
straight in testing instead?

Reply:
I use `Box<dyn AccountStorage>` in the `account_storage_tests.rs`.
And I think [AccountStorageBuilder::build](https://github.com/KomodoPlatform/atomicDEX-API/blob/gui-storage/mm2src/gui_storage/src/account/storage/mod.rs#L115) should return a boxed storage, not `SqliteAccountStorage` or `WasmAccountStorage`.
I think this approach is more practical and reliable in terms of new storage implementations.

Reply:
I guess It can be left as it is. It might even be used for something outside tests in the future, who knows.